Generative-AI/Cold-Email-Generator/app/app.py

```
@@ -45,6 +109,35 @@ def extract_jobs(self, cleaned_text):
+            If any information is not found, return empty string for strings or empty list for lists.
+            Only return the valid JSON.
+            ### VALID JSON (NO PREAMBLE):
+            """
+        )
+        chain_extract_user = prompt_extract_user | self.llm
+        res = chain_extract_user.invoke(input={"pdf_text": pdf_text})
+        try:
+            json_parser = JsonOutputParser()
+            res = json_parser.parse(res.content)
+        except OutputParserException:
+            raise OutputParserException("Unable to parse user information from PDF.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)